clash-lib/src/Clash/Normalize/Strategy.hs

```diff
@@ -117,7 +117,9 @@ constantPropagation =
     -- non-representable binder is inlined once, we can get rid of the recursive
     -- aspect using the case-of-known-constructor
     inlineNR :: NormRewrite
-    inlineNR = bottomupR (apply "inlineNonRep" inlineNonRep)
+    inlineNR =
+          bottomupR (apply "deadCode" deadCode)
+      >-! bottomupR (apply "inlineNonRep" inlineNonRep)
```

Comment:
Wouldn't it be more efficient to combine the two transformations in one bottom up traversal?

Reply:
We had some discussion on this on Slack, I'll update the commit message / PR message to explain the reasoning!

Reply:
I've updated the comment right above the transformation!